src/utils/resolveHOC.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ *
+ */
+
+import recast from 'recast';
+import isReactCreateClassCall from './isReactCreateClassCall';
+
+var {
+  types: {
+    NodePath,
+    namedTypes: types,
+  },
+} = recast;
+
+/**
+ * If the path is a call expression, it recursively resolves to the
+ * rightmost argument, stopping if it finds a React.createClass call expression
```

Comment:
Inspecting only the rightmost argument may not work for all cases.
One common case is for example Relay, where the component will be the [first argument](https://facebook.github.io/relay/docs/guides-containers.html#relay-containers).

Also it might be interesting to investigate `connect()` of [react-redux](https://github.com/reactjs/react-redux/blob/master/docs/api.md) and [recompose](https://github.com/acdlite/recompose/blob/master/docs/API.md), as a baseline for example usage of HoCs.